### PR TITLE
AVX-47161: Fix proxy profile update

### PR DESCRIPTION
**Pull request by avx-vjain**
- Removed ForceNew from all the fields that were optional
- Added ProxyProfile update functionality

### CODE REVIEW

File: aviatrix/resource_aviatrix_edge_proxy_profile_config.go [hunk 1]
```diff
@@ -8,9 +8,20 @@ import (
...
 )
 
+type proxyProfile struct {
+	Action      string  `json:"action"`
+	CID         string  `json:"CID"`
+	AccountName string  `json:"account_name"`
+	Name        string  `json:"proxy_name"`
+	Address     *string `json:"address"`
+	Port        *int    `json:"port"`
+	CACert      *string `json:"ca_cert"`
```
Comment (dnardo) [inline aviatrix/resource_aviatrix_edge_proxy_profile_config.go:18]: Why do these need to be pointers? Are we trying to determine  Zero Values and Unset Fields ?

File: aviatrix/resource_aviatrix_edge_proxy_profile_config.go [hunk 2]
```diff
@@ -70,6 +103,48 @@ func resourceAviatrixEdgeProxyProfileConfigCreate(d *schema.ResourceData, meta i
...
+		}
+		return fmt.Errorf("couldn't get proxy configuration: %s", err)
+	}
+
+	modifiedProxy := &goaviatrix.EdgePlatformProxyProfileUpdate{
+		EdgePlatformProxyProfile: goaviatrix.EdgePlatformProxyProfile{
+			AccountName: proxy.AccountName,
+			Name:        proxy.Name,
+			Address:     existingProxy.IPAddress,
+			Port:        int(existingProxy.Port),
```
Comment (dnardo) [inline aviatrix/resource_aviatrix_edge_proxy_profile_config.go]: Why don't we just avoid the conversion here and fix the struct in edge_platform_proxy_profile.go to be int. We know port won't be above 65535 and we know they will not have more than 65535 proxy profiles.  Seems simpler to me.

File: aviatrix/resource_aviatrix_edge_proxy_profile_config.go (hunk 1 above)
Comment (avx-vjain) [reply aviatrix/resource_aviatrix_edge_proxy_profile_config.go:18]: Unset fields

File: goaviatrix/edge_platform_proxy_profile.go [hunk 3]
```diff
@@ -15,16 +15,21 @@ type EdgePlatformProxyProfile struct {
...
-	ProxyProfileCount int64      `json:"deviceCount"`
-	CreatedAt         time.Time  `json:"createdAt"`
-	UpdatedAt         time.Time  `json:"updatedAt"`
-	CaCert            *string    `json:"caCert,omitempty"`
-	ExpiredAt         *time.Time `json:"expiredAt,omitempty"`
+	ProxyID           string    `json:"proxyID"`
+	Name              string    `json:"name"`
+	IPAddress         string    `json:"address"`
+	Port              int       `json:"port"`
+	ProxyProfileCount int64     `json:"deviceCount"`
```
Comment (dnardo) [inline goaviatrix/edge_platform_proxy_profile.go]: I'd make this an int also just so there is no conversion down the road.

**Review by dnardo — DISMISSED** [review]

**Review by dnardo — APPROVED** [review]

**Pull request merged**